lib/foreman_maintain/concerns/os_facts.rb

```diff
@@ -0,0 +1,54 @@
+module ForemanMaintain
+  module Concerns
+    module OsFacts
+      OS_RELEASE_FILE = '/etc/os-release'.freeze
+
+      def facts
+        unless defined?(@facts)
+          @facts = {}
+          tr_map = { /^#.*/ => '', '"' => '', "\n" => '' }
+          rexp = Regexp.union(tr_map.keys)
+          File.open(OS_RELEASE_FILE) do |file|
+            file.readlines.each do |line|
+              key, value = line.gsub(rexp, tr_map).split('=')
+              @facts[key] = value unless key.nil?
+            end
+          end
```

Comment:
I must admit I find the Rexexp solution hard to read, and would propose the following alternative

```suggestion
          File.open(OS_RELEASE_FILE) do |file|
            file.readlines.each do |line|
              line.strip! # drop any whitespace, including newlines from start and end of the line
              next if line.start_with?('#') # ignore comments
              key, value = line.split('=', 2) # split at most into 2 items, if the value ever contains an =
              @facts[key] = value.gsub(/"/, '') unless key.nil?
            end
          end
```

(not blocking tho, if you prefer your solution, I'm good with that)